### PR TITLE
Remove runtime functions from the stacktrace when runtime built with -g

### LINKED ISSUE #970: Remove runtime functions from the stacktrace when runtime built with -g
With #964 if the runtime is compiled with `-g` stack trace in the case of a panic will include the runtime functions (since runtime function calls now have locations). For example in the case of `01-int/div1-p.bal` we will get the stack trace as
```
panic: arithmetic overflow
    _bal_error_construct /home/heshan/work/nballerina/runtime/error.c:10
    _bal_panic_construct /home/heshan/work/nballerina/runtime/panic.c:67
    div ../../../compiler/testSuite/01-int/div1-p.bal
    main (root) ../../../compiler/testSuite/01-int/div1-p.bal:6
    main /home/heshan/work/nballerina/runtime/main.c:11
```
instead of 
```
panic: arithmetic overflow
    div ../../../compiler/testSuite/01-int/div1-p.bal:10
    main (root) ../../../compiler/testSuite/01-int/div1-p.bal:6
```

### PR DESCRIPTION
**Pull request by heshanpadmasiri**
## Purpose
Remove runtime functions from the stacktrace when runtime built with -g
Resolves #970

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: runtime/error.c [hunk 1]
```diff
@@ -61,6 +61,7 @@ static void processInitialPC(PC pc, BacktraceStartLine *backtraceStartLine);
 static enum DemangleResult demangle(const char *mangledName, const char **localName, FILE *fp);
 static bool demangleModules(const char **mangledModules, FILE *fp);
 static bool demangleCountedName(const char **pp, const char **name);
+static bool isPrefix(const char* prefix, const char* str);
```
Comment (jclark) [inline runtime/error.c]: Space before `*` not after.

File: runtime/error.c [hunk 2]
```diff
@@ -235,6 +240,18 @@ static void printBacktraceLine(const char *filename, int64_t lineNumber, const c
     fflush(fp);
 }
 
+static bool isPrefix(const char* prefix, const char* str) {
```
Comment (jclark) [inline runtime/error.c]: You can write this so that it looks at each character of prefix and str at most once. (Don't have calls to strlen.)

**Review by jclark — CHANGES_REQUESTED** [review]

File: runtime/error.c (hunk 1 above)
Comment (heshanpadmasiri) [reply runtime/error.c]: fixed

File: runtime/error.c (hunk 2 above)
Comment (heshanpadmasiri) [reply runtime/error.c]: fixed

File: runtime/error.c [hunk 3]
```diff
@@ -235,6 +240,15 @@ static void printBacktraceLine(const char *filename, int64_t lineNumber, const c
     fflush(fp);
 }
 
+static bool isPrefix(const char *prefix, const char *str) {
+    while (*++prefix == *++str) {
```
Comment (jclark) [inline runtime/error.c]: Don't think it's safe to ignore the first character.

```
for (; *prefix == *str && *prefix != '\0'; prefix++, str++) { }
```

**Review by jclark — CHANGES_REQUESTED** [review]

File: runtime/error.c (hunk 3 above)
Comment (heshanpadmasiri) [reply runtime/error.c]: fixed

**Pull request merged**